cross = "B_YJM145x.x.YPS163a"
allele1 = "YJM145x"
allele2 = "YPS163a"
genotypesIN = "data/genotype_B.tsv"
phenotypesIN = "data/phenotypes.tsv"
#defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "376_RMx-x-YPS163a"
#allele1 = "RMx"
#allele2 = "YPS163a"
#genotypesIN = "data/genotype_376.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "377_CLIB413a-x-YJM145x"
#allele1 = "CLIB413a"
#allele2 = "YJM145x"
#genotypesIN = "data/genotype_377.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "381_YJM454a-x-YJM978x"
#allele1 = "YJM454a"
#allele2 = "YJM978x"
#genotypesIN = "data/genotype_381.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "393_CLIB413a-x-YJM978x"
#allele1 = "CLIB413a"
#allele2 = "YJM978x"
#genotypesIN = "data/genotype_393.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3028_CLIB219x-x-M22"
#allele1 = "CLIB219x"
#allele2 = "M22"
#genotypesIN = "data/genotype_3028.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "2999_I14a-x-YPS1009x"
#allele1 = "I14a"
#allele2 = "YPS1009x"
#genotypesIN = "data/genotype_2999.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3000_I14a-x-Y10x"
#allele1 = "I14a"
#allele2 = "Y10x"
#genotypesIN = "data/genotype_3000.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3001_PW5a-x-Y10x"
#allele1 = "PW5a"
#allele2 = "Y10x"
#genotypesIN = "data/genotype_3001.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3003_273614xa-x-YJM981x"
#allele1 = "273614xa"
#allele2 = "YJM981x"
#genotypesIN = "data/genotype_3003.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3004_CBS2888a-x-YJM981x"
#allele1 = "CBS2888a"
#allele2 = "YJM981x"
#genotypesIN = "data/genotype_3004.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3043_CBS2888a-x-CLIB219x"
#allele1 = "CBS2888a"
#allele2 = "CLIB219x"
#genotypesIN = "data/genotype_3043.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

#cross = "3049_273614xa-x-PW5a"
#allele1 = "273614xa"
#allele2 = "PW5a"
#genotypesIN = "data/genotype_3049.tsv"
#phenotypesIN = "data/phenotypes.tsv"
##defined cross type as "haploid", based upon the R/qtl2 input file type definitions: https://kbroman.org/qtl2/assets/vignettes/input_files.html#Detailed_specifications_for_each_cross_type
#yaml = paste(cross,"_Lactate.Model1.yaml",sep="")

library(qtl2)
library(mutoss)
library(qvalue)

genoOUT = paste(cross,"_geno.csv",sep="")#simply use "A" and "B" notation, instead of varying this for each cross.
phenoOUT = paste(cross,"_pheno_Lactate.csv",sep="")
variantInfo = paste(cross,"_gmap.csv",sep="")# position can either be centiMorgans (cM) or Megabase pairs (Mbp).

##reformat files
print("##Reformatting Data##")
JB_geno.table = read.table(genotypesIN, head=T, sep="\t")
JB_pheno.table = read.table(phenotypesIN, head=T, sep="\t")
cross_samples = JB_geno.table$X

pheno_to_test = c("Lactate..1")
QTL2_pheno.table = JB_pheno.table[match(cross_samples, JB_pheno.table$id),]
QTL2_pheno.table = QTL2_pheno.table[,match(c("id",pheno_to_test), names(QTL2_pheno.table))]
write.csv(QTL2_pheno.table, phenoOUT, row.names=F, quote=F)

QTL2_geno.table = JB_geno.table
colnames(QTL2_geno.table)[1]="id"
convert_geno = function(arr){
	arr[arr == 1]="A"
	arr[arr == 2]="B"
	return(arr)
}#end def convert_geno
QTL2_geno.table[,2:ncol(QTL2_geno.table)]=apply(QTL2_geno.table[,2:ncol(QTL2_geno.table)], 2, convert_geno)
print(table(QTL2_geno.table$chrXIV_467219_A_G))#print genotype frequency for variant that I am trying to understand better
write.csv(QTL2_geno.table, genoOUT, row.names=F, quote=F)

extract_chr_and_pos = function(marker){
	#print(marker)
	temp.marker_info=unlist(strsplit(marker, split="_"))
	#print(temp.marker_info)
	temp.chr = temp.marker_info[1]
	temp.pos = temp.marker_info[2]
	return(c(temp.chr, temp.pos))
}#end def extract_chr
marker = colnames(QTL2_geno.table)[2:ncol(QTL2_geno.table)]
marker_info = t(sapply(marker, extract_chr_and_pos))
colnames(marker_info) = c("chr","pos")
QTL2_map.table = data.frame(marker, marker_info)
QTL2_map.table$pos = as.numeric(as.character(QTL2_map.table$pos))
QTL2_map.table$pos = QTL2_map.table$pos / 1000000
write.csv(QTL2_map.table, variantInfo, row.names=F, quote=F)

rm(JB_geno.table)
rm(JB_pheno.table)
rm(QTL2_geno.table)
rm(QTL2_map.table)
rm(QTL2_pheno.table)
rm(marker_info)
rm(marker)
rm(cross_samples)

print("##Re-Reading Data##")
cross_obj = read_cross2(yaml)

pheno = cross_obj$pheno[,pheno_to_test]

print("##Preprocessing, Similar to QTL##")
map = insert_pseudomarkers(cross_obj$gmap, step=1)
#follow code to use "error_prob=0.002" instead of "error_prob = 0.0001"
pr = calc_genoprob(cross_obj, map, error_prob=0.002)#includes parameter map_function = c("haldane", "kosambi", "c-f", "morgan")

print("##Kinship Estimation##")
kinship = calc_kinship(pr)
print("##Heritability Calculation##")
hsq = est_herit(pheno, kinship)
print(hsq[1])
print(hsq)
